### PR TITLE
[4942] Take Action Frontend component

### LINKED ISSUE #4942: [Frontend] Take action
Blocked until backend work https://github.com/mozilla/foundation.mozilla.org/issues/4842 is done.

Refer to Sketch file for the final design.

image

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #4942 

Preview
image

### CODE REVIEW

File: source/sass/components/take-action.scss [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  padding-bottom: 80px;
+}
+
+.take-action-title {
+  font-family: $font-family-sans-serif;
+  font-size: 23px;
+  font-weight: bold;
+  line-height: 1.3;
+  color: $black;
+  margin-bottom: 27px;
```
Comment (mmmavis) [inline source/sass/components/take-action.scss:12]: - Replace rules for font-size and font-height with `@include set-text-size(23px, 27px)`. 
- font color should already be black so `color: $black;` is not needed
- Bottom margin CSS rule can be replaced by applying a `mb-4` class to `take-action-title`

File: network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html [hunk 2]
```diff
@@ -105,34 +105,45 @@ <h2>{{ page.partner_heading }}</h2>
...
+      <div class="col-md-8">
+        <div class="row">
+          {% for item in page.take_action_cards.all %}
+            <div class="col-lg-6 d-flex align-items-stretch">
+              <div class="card mb-4">
+                {% image item.image fill-350x130 as img %}
+                  <img src="{{ img.url }}" class="card-img-top" alt="{{ img.alt }}">
+                  <div class="card-body">
+                    <a href="{{ item.internal_link.url }}" class="stretched-link take-action-link">
+                      <p class="card-text h5-heading">Some quick example text to build on the card title and make up the bulk of the card's content.</p>
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html:124]: This hardcoded placeholder text should be updated with content from the CMS.

File: source/sass/components/take-action.scss [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+.take-action-link:hover {
+  text-decoration: none;
```
Comment (mmmavis) [inline source/sass/components/take-action.scss:16]: This is not needed. Tais mentioned to mimic card design on the /blog page: https://github.com/mozilla/foundation.mozilla.org/issues/4942#issuecomment-666739257

File: source/js/components/join/join.jsx [hunk 4]
```diff
@@ -468,11 +468,7 @@ export default class JoinUs extends React.Component {
    * Render the submit button in signup CTA.
    */
   renderSubmitButton() {
-    let classnames = classNames("btn btn-primary", {
-      "w-100": !this.isFlowForm(),
-      "flex-1 mr-3": this.isFlowForm(),
-    });
-    return <button className={classnames}>{getText(`Sign up`)}</button>;
+    return <button className="btn btn-black px-4">{getText(`Sign up`)}</button>;
```
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/js/components/join/join.jsx:471]: Yay checked with designers and they are good with keeping the button the way it is (purple, like the rest of the form buttons)

https://mozilla.slack.com/archives/C4EGUBY2Z/p1596579392006300

File: source/js/components/petition/locale-strings.jsx [hunk 5]
```diff
@@ -48,9 +48,11 @@ export default {
       <React.Fragment>
         If you have already confirmed your opt-in to receive Mozilla-related
         emails, you can now{" "}
-        <a href="https://www.mozilla.org/newsletter/recovery/" target="_blank">
-          manage your subscriptions
-        </a>{" "}
+        <strong>
```
[the remark this answers is not in the document]
Comment (KalobTaulien) [reply source/js/components/petition/locale-strings.jsx:51]: It's not there, I didn't see this at all, but the link gets hidden in text so I followed the next closest link rule. 
image

This is the text that shows up after the form is submitted. Thoughts?

File: network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html (hunk 2 above)
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/homepage.html:124]: LOL This is why we do peer review, thank you for catching this!

**Comment by KalobTaulien** [comment]
@mmmavis I've made all your suggested changes, thank you for the pointing those out! 🎉  
Just one question in `locale-strings.jsx`

**Comment by KalobTaulien** [comment]
@mmmavis I rebased this branch on top of `ia-refresh-homepage` - could you re-review it?

File: source/js/components/join/join.jsx [hunk 6]
```diff
@@ -468,8 +468,7 @@ export default class JoinUs extends React.Component {
    * Render the submit button in signup CTA.
    */
   renderSubmitButton() {
-    let classnames = classNames("btn btn-primary", {
-      "w-100": !this.isFlowForm(),
+    let classnames = classNames("btn btn-secondary", {
```
Comment (mmmavis) [inline source/js/components/join/join.jsx:471]: Please revert this. Designers are fine with keepign the button as btn-primary. 👍

File: source/js/components/join/join.scss [hunk 7]
```diff
@@ -43,6 +43,16 @@
   .form-control {
     font-weight: 400;
     color: $black;
```
Comment (mmmavis) [inline source/js/components/join/join.scss:45]: Changes in this file can be reverted as well?

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/take_action.html [hunk 8]
```diff
@@ -1,28 +1,41 @@
...
+                  </div>
+              </div>
+            </div>
+          {% endfor %}
+        </div>
+      </div>
+      <div class="col-md-4 mb-md-4 d-flex align-items-stretch">
+        <div class="bg-black p-4 text-white">
+          <div
+            class="join-us dark-theme"
```
Comment (mmmavis) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/take_action.html:32]: let's remove `dark-theme` from here and add that class to line 30 instead

Comment (mmmavis) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/take_action.html:32]: Remove all `text-white` instances added in this PR. By applying `dark-theme` to the parent `div`, text color should be auto set to its lighter vesion.

File: source/js/components/petition/locale-strings.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (mmmavis) [reply source/js/components/petition/locale-strings.jsx:51]: We can revert changes in this file as well.

File: source/sass/components/take-action.scss [hunk 9]
```diff
@@ -0,0 +1,11 @@
+.take-action-section {
+  padding-top: 80px;
+  padding-bottom: 80px;
+}
+
+.take-action-title {
+  font-family: $font-family-sans-serif;
+  @include set-text-size(23px, 27px);
```
Comment (mmmavis) [inline source/sass/components/take-action.scss:8]: This should be `@include set-text-size(23px, 27px);`.

File: source/sass/components/take-action.scss [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+  font-weight: bold;
+  line-height: 1.3;
```
Comment (mmmavis) [inline source/sass/components/take-action.scss:10]: line 8 should take care of this 😁

**Comment by mmmavis** [comment]
@KalobTaulien it seems like this PR doesn't have the new fake data work? Can you rebase your branch again?

**Comment by taisdesouzalessa** [comment]
Could you send me the live link to see this component, please? Thanks!

**Comment by KalobTaulien** [comment]
> @KalobTaulien it seems like this PR doesn't have the new fake data work? Can you rebase your branch again?

Done! And rebased. Should be good to go. Let me know if you find anything else @mmmavis

**Comment by KalobTaulien** [comment]
> Could you send me the live link to see this component, please? Thanks!

@taisdesouzalessa [I think this is the link we're looking for](https://foundation-s-4942-take--poqspx.herokuapp.com/en/) (though I could be wrong)

**Comment by taisdesouzalessa** [comment]
@mmmavis what is the design review here for? Just the newsletter signup box or the whole "take action" component?  I ask because this is what I see on the [live link](https://foundation-s-4942-take--poqspx.herokuapp.com/en/):
image

**Comment by mmmavis** [comment]
Let me recreate the review app and see if it fixes the issue. Will update here once review app is live again.

**Comment by mmmavis** [comment]
@KalobTaulien one GitHub tip:

You can re-request PR review by clicking the refresh looking icon
image

GitHub will triage the PR to reviewer's PR review notification queue again. This way it's less easy for reviewers to miss the ping 😂 (like I often did)

**Comment by KalobTaulien** [comment]
@taisdesouzalessa I was wrong, it's [this link](https://foundation-s-4942-take--poqspx.herokuapp.com/en/)

**Comment by mmmavis** [comment]
K recreating the review app fixed the empty blocks issue: https://foundation-s-4942-take--hzsli6.herokuapp.com/en/

@taisdesouzalessa generally you can find review app for each PR using the "View deployment" button on this page or look for info posted in the Slack channel `mofo-ra-foundation`. But feel free to ping us if you aren't able to access it. Sometimes there are glitches that we need to manual fix so review app can spawn properly 🙈 
image

**Comment by taisdesouzalessa** [comment]
Ahh thanks for the "view deployment" button tip, Mavis :).

**Comment by KalobTaulien** [comment]
Thanks for the feedback @taisdesouzalessa! Ive made the adjustments you recommended. The review app should be done building and shared on Slack at some point today.

**Comment by Pomax** [comment]
@KalobTaulien for rebasing: rebase on `ia-refresh-homepage` and then during the rebase process, rename the migration file to 0013_... and update its dependency to point to the 0012_... file (as well as updating `pagemodels/base.py` to resolve the merge conflict), then `rebase --continue`, then once that's all done, _delete that new 0013_... migration file_ and rerun `inv makemigrations` to get a new, working, migration.

**Comment by KalobTaulien** [comment]
@taisdesouzalessa Thank you! I made your changes. 
@Pomax Thank you for these steps, I re-made the migrations and it seems to work perfectly for me now. 

Ready for last (minor) design review, and then code review from @mmmavis

**Comment by mmmavis** [comment]
PR updated! Merging it in now! 🎉 Good work @KalobTaulien

**Pull request merged**